Influxd: Compaction-related panic on HEAD-2016/04/26

During a compaction run (with the compaction intervals in the config file set to `5s`), `influxd` panic'ed. cc @toddboom @jwilder 

```
$ ./influxd run --config=influxdb.conf

 8888888           .d888 888                   8888888b.  888888b.
   888            d88P"  888                   888  "Y88b 888  "88b
   888            888    888                   888    888 888  .88P
   888   88888b.  888888 888 888  888 888  888 888    888 8888888K.
   888   888 "88b 888    888 888  888  Y8bd8P' 888    888 888  "Y88b
   888   888  888 888    888 888  888   X88K   888    888 888    888
   888   888  888 888    888 Y88b 888 .d8""8b. 888  .d88P 888   d88P
 8888888 888  888 888    888  "Y88888 888  888 8888888P"  8888888P"

[run] 2016/05/10 16:13:04 InfluxDB starting, version 0.10.0, branch master, commit 662ead753b56dacee72f67d093d920729f73a3da
[run] 2016/05/10 16:13:04 Go version go1.6.2, GOMAXPROCS set to 8
[run] 2016/05/10 16:13:04 Using configuration at: influxdb.conf
[store]2016/05/10 16:13:04 Using data dir: /mnt/nvme/influxdb_data/db/data
[tsm1wal] 2016/05/10 16:13:04 tsm1 WAL starting with 10485760 segment size
[tsm1wal] 2016/05/10 16:13:04 tsm1 WAL writing to /mnt/nvme/influxdb_data/db/wal/_internal/monitor/1
[tsm1wal] 2016/05/10 16:13:04 tsm1 WAL starting with 10485760 segment size
[tsm1wal] 2016/05/10 16:13:04 tsm1 WAL writing to /mnt/nvme/influxdb_data/db/wal/benchmark_db/default/2
[cacheloader] 2016/05/10 16:13:04 reading file /mnt/nvme/influxdb_data/db/wal/_internal/monitor/1/_00001.wal, size 55128
[filestore] 2016/05/10 16:13:04 /mnt/nvme/influxdb_data/db/data/benchmark_db/default/2/000000073-000000001.tsm (#4) opened in 5.758746ms
[filestore] 2016/05/10 16:13:04 /mnt/nvme/influxdb_data/db/data/benchmark_db/default/2/000000072-000000003.tsm (#3) opened in 5.034206ms
[filestore] 2016/05/10 16:13:04 /mnt/nvme/influxdb_data/db/data/benchmark_db/default/2/000000068-000000003.tsm (#2) opened in 5.738256ms
[filestore] 2016/05/10 16:13:04 /mnt/nvme/influxdb_data/db/data/benchmark_db/default/2/000000064-000000004.tsm (#1) opened in 5.708621ms
[filestore] 2016/05/10 16:13:04 /mnt/nvme/influxdb_data/db/data/benchmark_db/default/2/000000048-000000005.tsm (#0) opened in 7.144912ms
[cacheloader] 2016/05/10 16:13:04 reading file /mnt/nvme/influxdb_data/db/wal/benchmark_db/default/2/_00343.wal, size 10871632
[cacheloader] 2016/05/10 16:13:04 reading file /mnt/nvme/influxdb_data/db/wal/_internal/monitor/1/_00002.wal, size 0
[tsm1] 2016/05/10 16:13:04 /mnt/nvme/influxdb_data/db/data/_internal/monitor/1 database index loaded in 2.680021ms
[store]2016/05/10 16:13:04 /mnt/nvme/influxdb_data/db/data/_internal/monitor/1 opened in 20.046772ms
[cacheloader] 2016/05/10 16:13:04 reading file /mnt/nvme/influxdb_data/db/wal/benchmark_db/default/2/_00344.wal, size 722946
[cacheloader] 2016/05/10 16:13:04 reading file /mnt/nvme/influxdb_data/db/wal/benchmark_db/default/2/_00345.wal, size 0
[tsm1] 2016/05/10 16:13:05 /mnt/nvme/influxdb_data/db/data/benchmark_db/default/2 database index loaded in 527.878343ms
[store]2016/05/10 16:13:05 /mnt/nvme/influxdb_data/db/data/benchmark_db/default/2 opened in 959.814821ms
[subscriber] 2016/05/10 16:13:05 opened service
[monitor] 2016/05/10 16:13:05 Starting monitor system
[monitor] 2016/05/10 16:13:05 'build' registered for diagnostics monitoring
[monitor] 2016/05/10 16:13:05 'runtime' registered for diagnostics monitoring
[monitor] 2016/05/10 16:13:05 'network' registered for diagnostics monitoring
[monitor] 2016/05/10 16:13:05 'system' registered for diagnostics monitoring
[cluster] 2016/05/10 16:13:05 Starting cluster service
[shard-precreation] 2016/05/10 16:13:05 Starting precreation service with check interval of 10m0s, advance period of 30m0s
[snapshot] 2016/05/10 16:13:05 Starting snapshot service
[copier] 2016/05/10 16:13:05 Starting copier service
[admin] 2016/05/10 16:13:05 Starting admin service
[monitor] 2016/05/10 16:13:05 Storing statistics in database '_internal' retention policy 'monitor', at interval 10s
[admin] 2016/05/10 16:13:05 Listening on HTTP: [::]:8083
[continuous_querier] 2016/05/10 16:13:05 Starting continuous query service
[httpd]2016/05/10 16:13:05 Starting HTTP service
[httpd]2016/05/10 16:13:05 Authentication enabled: false
[httpd]2016/05/10 16:13:05 Listening on HTTP: [::]:8086
[retention] 2016/05/10 16:13:05 Starting retention policy enforcement service with check interval of 30m0s
[run] 2016/05/10 16:13:05 Listening for signals
2016/05/10 16:13:05 Sending anonymous usage statistics to m.influxdb.com
[tsm1] 2016/05/10 16:13:09 beginning full compaction of group 0, 5 TSM files
[tsm1] 2016/05/10 16:13:09 compacting full group (0) /mnt/nvme/influxdb_data/db/data/benchmark_db/default/2/000000048-000000005.tsm (#0)
[tsm1] 2016/05/10 16:13:09 compacting full group (0) /mnt/nvme/influxdb_data/db/data/benchmark_db/default/2/000000064-000000004.tsm (#1)
[tsm1] 2016/05/10 16:13:09 compacting full group (0) /mnt/nvme/influxdb_data/db/data/benchmark_db/default/2/000000068-000000003.tsm (#2)
[tsm1] 2016/05/10 16:13:09 compacting full group (0) /mnt/nvme/influxdb_data/db/data/benchmark_db/default/2/000000072-000000003.tsm (#3)
[tsm1] 2016/05/10 16:13:09 compacting full group (0) /mnt/nvme/influxdb_data/db/data/benchmark_db/default/2/000000073-000000001.tsm (#4)
[tsm1] 2016/05/10 16:13:10 beginning level 1 compaction of group 0, 2 TSM files
[tsm1] 2016/05/10 16:13:10 compacting level 1 group (0) /mnt/nvme/influxdb_data/db/data/benchmark_db/default/2/000000073-000000001.tsm (#0)
[tsm1] 2016/05/10 16:13:10 compacting level 1 group (0) /mnt/nvme/influxdb_data/db/data/benchmark_db/default/2/000000075-000000001.tsm (#1)
[tsm1] 2016/05/10 16:13:11 compacted level 1 group (0) into /mnt/nvme/influxdb_data/db/data/benchmark_db/default/2/000000075-000000002.tsm.tmp (#0)
[tsm1] 2016/05/10 16:13:11 compacted level 1 group 0 of 2 files into 1 files in 61.043111ms
[tsm1] 2016/05/10 16:13:21 beginning level 1 compaction of group 0, 2 TSM files
[tsm1] 2016/05/10 16:13:21 compacting level 1 group (0) /mnt/nvme/influxdb_data/db/data/_internal/monitor/1/000000001-000000001.tsm (#0)
[tsm1] 2016/05/10 16:13:21 compacting level 1 group (0) /mnt/nvme/influxdb_data/db/data/_internal/monitor/1/000000002-000000001.tsm (#1)
[tsm1] 2016/05/10 16:13:21 beginning full compaction of group 0, 2 TSM files
[tsm1] 2016/05/10 16:13:21 compacting full group (0) /mnt/nvme/influxdb_data/db/data/_internal/monitor/1/000000001-000000001.tsm (#0)
[tsm1] 2016/05/10 16:13:21 compacting full group (0) /mnt/nvme/influxdb_data/db/data/_internal/monitor/1/000000002-000000001.tsm (#1)
[tsm1] 2016/05/10 16:13:21 compacted level 1 group (0) into /mnt/nvme/influxdb_data/db/data/_internal/monitor/1/000000002-000000002.tsm.tmp (#0)
[tsm1] 2016/05/10 16:13:21 compacted level 1 group 0 of 2 files into 1 files in 2.402858ms
[tsm1] 2016/05/10 16:13:21 error replacing new TSM files: rename /mnt/nvme/influxdb_data/db/data/_internal/monitor/1/000000002-000000002.tsm.tmp /mnt/nvme/influxdb_data/db/data/_internal/monitor/1/000000002-000000002.tsm: no such file or directory
[tsm1] 2016/05/10 16:13:32 compacted full group (0) into /mnt/nvme/influxdb_data/db/data/benchmark_db/default/2/000000073-000000002.tsm.tmp (#0)
[tsm1] 2016/05/10 16:13:32 compacted full 5 files into 1 files in 22.734636333s
[tsm1] 2016/05/10 16:13:32 beginning level 2 compaction of group 0, 2 TSM files
[tsm1] 2016/05/10 16:13:32 compacting level 2 group (0) /mnt/nvme/influxdb_data/db/data/benchmark_db/default/2/000000073-000000002.tsm (#0)
[tsm1] 2016/05/10 16:13:32 compacting level 2 group (0) /mnt/nvme/influxdb_data/db/data/benchmark_db/default/2/000000075-000000002.tsm (#1)
[tsm1] 2016/05/10 16:13:41 beginning level 1 compaction of group 0, 2 TSM files
[tsm1] 2016/05/10 16:13:41 compacting level 1 group (0) /mnt/nvme/influxdb_data/db/data/_internal/monitor/1/000000003-000000001.tsm (#0)
[tsm1] 2016/05/10 16:13:41 compacting level 1 group (0) /mnt/nvme/influxdb_data/db/data/_internal/monitor/1/000000004-000000001.tsm (#1)
[tsm1] 2016/05/10 16:13:41 compacted level 1 group (0) into /mnt/nvme/influxdb_data/db/data/_internal/monitor/1/000000004-000000002.tsm.tmp (#0)
[tsm1] 2016/05/10 16:13:41 compacted level 1 group 0 of 2 files into 1 files in 5.079504ms
[tsm1] 2016/05/10 16:13:42 beginning level 2 compaction of group 0, 2 TSM files
[tsm1] 2016/05/10 16:13:42 compacting level 2 group (0) /mnt/nvme/influxdb_data/db/data/_internal/monitor/1/000000002-000000002.tsm (#0)
[tsm1] 2016/05/10 16:13:42 compacting level 2 group (0) /mnt/nvme/influxdb_data/db/data/_internal/monitor/1/000000004-000000002.tsm (#1)
panic: runtime error: slice bounds out of range

goroutine 98 [running]:
panic(0xb4a800, 0xc82000e090)
        /usr/local/go/src/runtime/panic.go:481 +0x3e6
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*indirectIndex).UnmarshalBinary(0xc823764080, 0x7fb7df834a15, 0x2b90, 0x2b98, 0x0, 0x0)
        /Users/todd/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/reader.go:671 +0x515
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*mmapAccessor).init(0xc824c3c140, 0x0, 0x0, 0x0, 0x0)
        /Users/todd/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/reader.go:981 +0x569
github.com/influxdata/influxdb/tsdb/engine/tsm1.NewTSMReaderWithOptions(0x0, 0x0, 0xc820960010, 0x0, 0x0, 0x0)
        /Users/todd/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/reader.go:157 +0x2df
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Compactor).compact(0xc8201cc090, 0xc820080701, 0xc824b40220, 0x2, 0x2, 0x0, 0x0, 0x0, 0x0, 0x0)
        /Users/todd/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/compact.go:459 +0x306
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Compactor).CompactFast(0xc8201cc090, 0xc824b40220, 0x2, 0x2, 0x0, 0x0, 0x0, 0x0, 0x0)
        /Users/todd/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/compact.go:486 +0x71
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Engine).compactTSMLevel.func1(0xc820074400, 0xc82011b600, 0x2, 0x1, 0x0, 0xc824b40220, 0x2, 0x2)
        /Users/todd/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/engine.go:577 +0x5fd
created by github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Engine).compactTSMLevel
        /Users/todd/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/engine.go:603 +0x265

```

Also the database is corrupted. When restarting `influxd` this happens:

```
$ ./influxd run --config=influxdb.conf

 8888888           .d888 888                   8888888b.  888888b.
   888            d88P"  888                   888  "Y88b 888  "88b
   888            888    888                   888    888 888  .88P
   888   88888b.  888888 888 888  888 888  888 888    888 8888888K.
   888   888 "88b 888    888 888  888  Y8bd8P' 888    888 888  "Y88b
   888   888  888 888    888 888  888   X88K   888    888 888    888
   888   888  888 888    888 Y88b 888 .d8""8b. 888  .d88P 888   d88P
 8888888 888  888 888    888  "Y88888 888  888 8888888P"  8888888P"

[run] 2016/05/10 16:20:43 InfluxDB starting, version 0.10.0, branch master, commit 662ead753b56dacee72f67d093d920729f73a3da
[run] 2016/05/10 16:20:43 Go version go1.6.2, GOMAXPROCS set to 8
[run] 2016/05/10 16:20:43 Using configuration at: influxdb.conf
[store]2016/05/10 16:20:43 Using data dir: /mnt/nvme/influxdb_data/db/data
[tsm1wal] 2016/05/10 16:20:43 tsm1 WAL starting with 10485760 segment size
[tsm1wal] 2016/05/10 16:20:43 tsm1 WAL writing to /mnt/nvme/influxdb_data/db/wal/_internal/monitor/1
[tsm1wal] 2016/05/10 16:20:43 tsm1 WAL starting with 10485760 segment size
[tsm1wal] 2016/05/10 16:20:43 tsm1 WAL writing to /mnt/nvme/influxdb_data/db/wal/benchmark_db/default/2
[filestore] 2016/05/10 16:20:43 /mnt/nvme/influxdb_data/db/data/_internal/monitor/1/000000004-000000002.tsm (#1) opened in 109.476µs
panic: runtime error: slice bounds out of range

goroutine 21 [running]:
panic(0xb4a800, 0xc82000e090)
        /usr/local/go/src/runtime/panic.go:481 +0x3e6
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*indirectIndex).UnmarshalBinary(0xc8201f2000, 0x7f49f5913a15, 0x2b90, 0x2b98, 0x0, 0x0)
        /Users/todd/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/reader.go:671 +0x515
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*mmapAccessor).init(0xc8201e0050, 0x0, 0x0, 0x0, 0x0)
        /Users/todd/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/reader.go:981 +0x569
github.com/influxdata/influxdb/tsdb/engine/tsm1.NewTSMReaderWithOptions(0x0, 0x0, 0xc82007e080, 0x0, 0x0, 0x0)
        /Users/todd/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/reader.go:157 +0x2df
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*FileStore).Open.func1(0xc82009c380, 0xc8200725a0, 0x0, 0xc82007e080)
        /Users/todd/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/file_store.go:325 +0x9b
created by github.com/influxdata/influxdb/tsdb/engine/tsm1.(*FileStore).Open
        /Users/todd/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/file_store.go:335 +0x749
```

I don't know if it's related. I have also compactor issue, and influxdb won't restart:
`
[run] 2016/05/13 08:32:20 InfluxDB starting, version 0.13.0~rc2, branch master, commit 590b3f9fc76904c0035ef1e9731ad1c27903ef76
[run] 2016/05/13 08:32:20 Go version go1.6.2, GOMAXPROCS set to 24
[run] 2016/05/13 08:32:20 Using configuration at: /etc/influxdb.toml
[store] 2016/05/13 08:32:20 Using data dir: /data/data
[tsm1wal] 2016/05/13 08:32:20 tsm1 WAL starting with 10485760 segment size
[tsm1wal] 2016/05/13 08:32:20 tsm1 WAL writing to /data/wal/_internal/monitor/1
[tsm1wal] 2016/05/13 08:32:20 tsm1 WAL starting with 10485760 segment size
[tsm1wal] 2016/05/13 08:32:20 tsm1 WAL writing to /data/wal/_internal/monitor/3
[cacheloader] 2016/05/13 08:32:20 reading file /data/wal/_internal/monitor/3/_00001.wal, size 1700879
[filestore] 2016/05/13 08:32:20 /data/data/_internal/monitor/1/000000001-000000001.tsm (#0) opened in 122.131µs
[cacheloader] 2016/05/13 08:32:20 reading file /data/wal/_internal/monitor/1/_00004.wal, size 0
[tsm1wal] 2016/05/13 08:32:20 tsm1 WAL starting with 10485760 segment size
[tsm1wal] 2016/05/13 08:32:20 tsm1 WAL writing to /data/wal/results/default/2
[shard] 2016/05/13 08:32:20 /data/data/_internal/monitor/1 database index loaded in 4.051656ms
[store] 2016/05/13 08:32:20 /data/data/_internal/monitor/1 opened in 6.408679ms
[filestore] 2016/05/13 08:32:20 /data/data/results/default/2/000000032-000000004.tsm (#1) opened in 27.552789ms
[filestore] 2016/05/13 08:32:20 /data/data/results/default/2/000000016-000000004.tsm (#0) opened in 29.453147ms
[cacheloader] 2016/05/13 08:32:20 reading file /data/wal/results/default/2/_00065.wal, size 56193
[cacheloader] 2016/05/13 08:32:20 reading file /data/wal/results/default/2/_00066.wal, size 0
[tsm1] 2016/05/13 08:32:20 beginning full compaction of group 0, 2 TSM files
[tsm1] 2016/05/13 08:32:20 compacting full group (0) /data/data/results/default/2/000000016-000000004.tsm (#0)
[tsm1] 2016/05/13 08:32:20 compacting full group (0) /data/data/results/default/2/000000032-000000004.tsm (#1)
[cacheloader] 2016/05/13 08:32:20 reading file /data/wal/_internal/monitor/3/_00002.wal, size 0
[shard] 2016/05/13 08:32:20 /data/data/_internal/monitor/3 database index loaded in 862.128µs
[store] 2016/05/13 08:32:20 /data/data/_internal/monitor/3 opened in 182.801127ms
panic: interface conversion: tsm1.Value is *tsm1.StringValue, not *tsm1.FloatValue

goroutine 35 [running]:
panic(0xb68180, 0xc821cd7ec0)
	/usr/local/go/src/runtime/panic.go:481 +0x3e6
github.com/influxdata/influxdb/tsdb/engine/tsm1.encodeFloatBlock(0x0, 0x0, 0x0, 0xc8223d2500, 0x3e8, 0x450, 0x0, 0x0, 0x0, 0x0, ...)
	/root/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/encoding.go:367 +0x8cd
github.com/influxdata/influxdb/tsdb/engine/tsm1.Values.Encode(0xc8223d2500, 0x3e8, 0x450, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/root/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/encoding.go:193 +0x408
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*tsmKeyIterator).chunk(0xc8200e2300, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/root/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/compact.go:974 +0xe3
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*tsmKeyIterator).combine(0xc8200e2300, 0xc82003a300, 0x0, 0x0, 0x0)
	/root/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/compact.go:965 +0xd0a
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*tsmKeyIterator).merge(0xc8200e2300)
	/root/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/compact.go:845 +0x13f
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*tsmKeyIterator).Next(0xc8200e2300, 0xc800437cff)
	/root/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/compact.go:813 +0xbe5
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Compactor).write(0xc82019b530, 0xc8206e2100, 0x38, 0x7f2a2fbfe798, 0xc8200e2300, 0x0, 0x0)
	/root/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/compact.go:560 +0x53b
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Compactor).writeNewFiles(0xc82019b530, 0x20, 0x5, 0x7f2a2fbfe798, 0xc8200e2300, 0x0, 0x0, 0x0, 0x0, 0x0)
	/root/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/compact.go:507 +0x3a1
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Compactor).compact(0xc82019b530, 0xc82000db00, 0xc8208b41c0, 0x2, 0x2, 0x0, 0x0, 0x0, 0x0, 0x0)
	/root/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/compact.go:473 +0x4fe
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Compactor).CompactFull(0xc82019b530, 0xc8208b41c0, 0x2, 0x2, 0x0, 0x0, 0x0, 0x0, 0x0)
	/root/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/compact.go:478 +0x71
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Engine).compactTSMFull.func1(0xc820218e70, 0xc82011fad0, 0x0, 0xc8208b41c0, 0x2, 0x2)
	/root/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/engine.go:754 +0x4c6
created by github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Engine).compactTSMFull
	/root/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/engine.go:772 +0x27e
`

FWIW, we got this moving 

We ahve this in the log:

```
[tsm1] 2016/05/16 16:38:25 beginning full compaction of group 0, 2 TSM files
```

Full log at https://gist.github.com/daviesalex/36f3fe30a433dd6a27b2f60cf1598e7f

Panic for us:

```
[retention] 2016/05/16 16:40:58 Starting retention policy enforcement service with check interval of 10m0s
[udp] 2016/05/16 16:40:58 Started listening on UDP: :8089
[udp] 2016/05/16 16:40:58 Started listening on UDP: :8090
[run] 2016/05/16 16:40:58 Listening for signals
panic: interface conversion: tsm1.Value is *tsm1.FloatValue, not *tsm1.IntegerValue

goroutine 1852 [running]:
panic(0xb76bc0, 0xc936def740)
    /usr/local/go/src/runtime/panic.go:481 +0x3e6
github.com/influxdata/influxdb/tsdb/engine/tsm1.encodeIntegerBlock(0x0, 0x0, 0x0, 0xc95ec7c000, 0x2ff, 0x400, 0x0, 0x0, 0x0, 0x0, ...)
    /root/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/encoding.go:771 +0x72b
github.com/influxdata/influxdb/tsdb/engine/tsm1.Values.Encode(0xc95ec7c000, 0x2ff, 0x400, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
    /root/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/encoding.go:195 +0x22c
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*tsmKeyIterator).chunk(0xc8ca12ef00, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
    /root/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/compact.go:992 +0x4af
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*tsmKeyIterator).combine(0xc8ca12ef00, 0xc986cd9400, 0x0, 0x0, 0x0)
    /root/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/compact.go:965 +0xd0a
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*tsmKeyIterator).merge(0xc8ca12ef00)
    /root/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/compact.go:845 +0x13f
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*tsmKeyIterator).Next(0xc8ca12ef00, 0xc83e5d0531)
    /root/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/compact.go:813 +0xbe5
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Compactor).write(0xc822aa8ab0, 0xc926539630, 0x4a, 0x7fe690183820, 0xc8ca12ef00, 0x0, 0x0)
    /root/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/compact.go:560 +0x53b
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Compactor).writeNewFiles(0xc822aa8ab0, 0x212, 0x5, 0x7fe690183820, 0xc8ca12ef00, 0x0, 0x0, 0x0, 0x0, 0x0)
    /root/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/compact.go:507 +0x3a1
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Compactor).compact(0xc822aa8ab0, 0xc8222e2500, 0xc9a352dfc0, 0x2, 0x2, 0x0, 0x0, 0x0, 0x0, 0x0)
    /root/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/compact.go:473 +0x4fe
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Compactor).CompactFull(0xc822aa8ab0, 0xc9a352dfc0, 0x2, 0x2, 0x0, 0x0, 0x0, 0x0, 0x0)
    /root/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/compact.go:478 +0x71
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Engine).compactTSMFull.func1(0xc966b63f60, 0xc820124580, 0x0, 0xc9a352dfc0, 0x2, 0x2)
    /root/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/engine.go:760 +0x4c6
created by github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Engine).compactTSMFull
    /root/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/engine.go:778 +0x27e
```

I am testing a downgrade from 0.14.0~n201605160800  to 0.13 stable. Thats starting now, will report back...

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -103,6 +103,7 @@ With this release the systemd configuration files for InfluxDB will use the syst
 - [#6946](https://github.com/influxdata/influxdb/issues/6946): Duplicate data for the same timestamp
 - [#7043](https://github.com/influxdata/influxdb/pull/7043): Remove limiter from walkShards
 - [#5501](https://github.com/influxdata/influxdb/issues/5501): Queries against files that have just been compacted need to point to new files
+- [#6595](https://github.com/influxdata/influxdb/issues/6595): Fix full compactions conflicting with level compactions
 
 ## v0.13.0 [2016-05-12]
 

--- a/tsdb/engine/tsm1/compact.go
+++ b/tsdb/engine/tsm1/compact.go
@@ -32,10 +32,11 @@ const (
 )
 
 var (
-	errMaxFileExceeded     = fmt.Errorf("max file exceeded")
-	errSnapshotsDisabled   = fmt.Errorf("snapshots disabled")
-	errCompactionsDisabled = fmt.Errorf("compactions disabled")
-	errCompactionAborted   = fmt.Errorf("compaction aborted")
+	errMaxFileExceeded      = fmt.Errorf("max file exceeded")
+	errSnapshotsDisabled    = fmt.Errorf("snapshots disabled")
+	errCompactionsDisabled  = fmt.Errorf("compactions disabled")
+	errCompactionAborted    = fmt.Errorf("compaction aborted")
+	errCompactionInProgress = fmt.Errorf("compaction in progress")
 )
 
 var (
@@ -695,13 +696,9 @@ func (c *Compactor) writeNewFiles(generation, sequence int, iter KeyIterator) ([
 }
 
 func (c *Compactor) write(path string, iter KeyIterator) (err error) {
-	if _, err := os.Stat(path); !os.IsNotExist(err) {
-		return fmt.Errorf("%v already file exists. aborting", path)
-	}
-
-	fd, err := os.OpenFile(path, os.O_CREATE|os.O_RDWR, 0666)
+	fd, err := os.OpenFile(path, os.O_CREATE|os.O_RDWR|os.O_EXCL, 0666)
 	if err != nil {
-		return err
+		return errCompactionInProgress
 	}
 
 	// Create the write for the new TSM file.

--- a/tsdb/engine/tsm1/compact.go
+++ b/tsdb/engine/tsm1/compact.go
@@ -492,6 +492,7 @@ type Compactor struct {
 	mu      sync.RWMutex
 	opened  bool
 	closing chan struct{}
+	files   map[string]struct{}
 }
 
 func (c *Compactor) Open() {
@@ -503,6 +504,7 @@ func (c *Compactor) Open() {
 
 	c.closing = make(chan struct{})
 	c.opened = true
+	c.files = make(map[string]struct{})
 }
 
 func (c *Compactor) Close() {
@@ -604,6 +606,11 @@ func (c *Compactor) CompactFull(tsmFiles []string) ([]string, error) {
 		return nil, errCompactionsDisabled
 	}
 
+	if !c.add(tsmFiles) {
+		return nil, errCompactionInProgress
+	}
+	defer c.remove(tsmFiles)
+
 	files, err := c.compact(false, tsmFiles)
 
 	// See if we were closed while writing a snapshot
@@ -627,6 +634,11 @@ func (c *Compactor) CompactFast(tsmFiles []string) ([]string, error) {
 	if !opened {
 		return nil, errCompactionsDisabled
 	}
+
+	if !c.add(tsmFiles) {
+		return nil, errCompactionInProgress
+	}
+	defer c.remove(tsmFiles)
 
 	files, err := c.compact(true, tsmFiles)
 
@@ -673,9 +685,6 @@ func (c *Compactor) writeNewFiles(generation, sequence int, iter KeyIterator) ([
 
 		// We hit an error but didn't finish the compaction.  Remove the temp file and abort.
 		if err != nil {
-			if err := os.Remove(fileName); err != nil {
-				return nil, err
-			}
 			return nil, err
 		}
 
@@ -748,6 +757,36 @@ func (c *Compactor) write(path string, iter KeyIterator) (err error) {
 		return err
 	}
 	return nil
+}
+
+func (c *Compactor) add(files []string) bool {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	var inuse bool
+	for _, f := range files {
+		if _, ok := c.files[f]; ok {
+			inuse = true
+			break
+		}
+	}
+
+	if inuse {
+		return false
+	}
+
+	for _, f := range files {
+		c.files[f] = struct{}{}
+	}
+	return true
+}
+
+func (c *Compactor) remove(files []string) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	for _, f := range files {
+		delete(c.files, f)
+	}
 }
 
 // KeyIterator allows iteration over set of keys and values in sorted order.

--- a/tsdb/engine/tsm1/compact.go
+++ b/tsdb/engine/tsm1/compact.go
@@ -763,18 +763,14 @@ func (c *Compactor) add(files []string) bool {
 	c.mu.Lock()
 	defer c.mu.Unlock()
 
-	var inuse bool
+	// See if the new files are already in use
 	for _, f := range files {
 		if _, ok := c.files[f]; ok {
-			inuse = true
-			break
+			return false
 		}
 	}
 
-	if inuse {
-		return false
-	}
-
+	// Mark all the new files in use
 	for _, f := range files {
 		c.files[f] = struct{}{}
 	}

--- a/tsdb/engine/tsm1/compact.go
+++ b/tsdb/engine/tsm1/compact.go
@@ -71,11 +71,6 @@ type DefaultPlanner struct {
 	// should always be greater than the CacheFlushWriteColdDuraion
 	CompactFullWriteColdDuration time.Duration
 
-	// lastPlanCompactedFull will be true if the last time
-	// Plan was called, all files were over the max size
-	// or there was only one file
-	lastPlanCompactedFull bool
-
 	// lastPlanCheck is the last time Plan was called
 	lastPlanCheck time.Time
 }
@@ -286,7 +281,7 @@ func (c *DefaultPlanner) Plan(lastWrite time.Time) []CompactionGroup {
 	generations := c.findGenerations()
 
 	// first check if we should be doing a full compaction because nothing has been written in a long time
-	if !c.lastPlanCompactedFull && c.CompactFullWriteColdDuration > 0 && time.Now().Sub(lastWrite) > c.CompactFullWriteColdDuration && len(generations) > 1 {
+	if c.CompactFullWriteColdDuration > 0 && time.Now().Sub(lastWrite) > c.CompactFullWriteColdDuration && len(generations) > 1 {
 		var tsmFiles []string
 		for i, group := range generations {
 			var skip bool
@@ -315,8 +310,6 @@ func (c *DefaultPlanner) Plan(lastWrite time.Time) []CompactionGroup {
 			}
 		}
 		sort.Strings(tsmFiles)
-
-		c.lastPlanCompactedFull = true
 
 		if len(tsmFiles) <= 1 {
 			return nil
@@ -455,8 +448,6 @@ func (c *DefaultPlanner) Plan(lastWrite time.Time) []CompactionGroup {
 		sort.Strings(cGroup)
 		tsmFiles = append(tsmFiles, cGroup)
 	}
-
-	c.lastPlanCompactedFull = false
 
 	return tsmFiles
 }

--- a/tsdb/engine/tsm1/compact_test.go
+++ b/tsdb/engine/tsm1/compact_test.go
@@ -1860,12 +1860,8 @@ func TestDefaultPlanner_Plan_SkipPlanningAfterFull(t *testing.T) {
 	if exp, got := 0, len(cp.Plan(time.Now().Add(-time.Second))); got != exp {
 		t.Fatalf("tsm file length mismatch: got %v, exp %v", got, exp)
 	}
-	// even though we do this, the planner should remember that last time we were over
-	cp.FileStore = fs
-	if exp, got := 0, len(cp.Plan(time.Now().Add(-time.Second))); got != exp {
-		t.Fatalf("tsm file length mismatch: got %v, exp %v", got, exp)
-	}
 
+	cp.FileStore = fs
 	// ensure that it will plan if last modified has changed
 	fs.lastModified = time.Now()
 

--- a/tsdb/engine/tsm1/engine.go
+++ b/tsdb/engine/tsm1/engine.go
@@ -858,7 +858,15 @@ func (e *Engine) compactTSMLevel(fast bool, level int) {
 
 					if fast {
 						files, err = e.Compactor.CompactFast(group)
-						if err != nil && err != errCompactionsDisabled {
+						if err == errCompactionsDisabled || err == errCompactionInProgress {
+							e.logger.Printf("aborted level %d group (%d). %v",
+								level, groupNum, err)
+
+							if err == errCompactionInProgress {
+								time.Sleep(time.Second)
+							}
+							return
+						} else if err != nil {
 							e.logger.Printf("error compacting TSM files: %v", err)
 							atomic.AddInt64(&e.stats.TSMCompactionErrors[level-1], 1)
 							time.Sleep(time.Second)
@@ -866,7 +874,15 @@ func (e *Engine) compactTSMLevel(fast bool, level int) {
 						}
 					} else {
 						files, err = e.Compactor.CompactFull(group)
-						if err != nil && err != errCompactionsDisabled {
+						if err == errCompactionsDisabled || err == errCompactionInProgress {
+							e.logger.Printf("aborted level %d compaction group (%d). %v",
+								level, groupNum, err)
+
+							if err == errCompactionInProgress {
+								time.Sleep(time.Second)
+							}
+							return
+						} else if err != nil {
 							e.logger.Printf("error compacting TSM files: %v", err)
 							atomic.AddInt64(&e.stats.TSMCompactionErrors[level-1], 1)
 							time.Sleep(time.Second)
@@ -941,7 +957,15 @@ func (e *Engine) compactTSMFull() {
 					)
 					if optimize {
 						files, err = e.Compactor.CompactFast(group)
-						if err != nil && err != errCompactionsDisabled {
+						if err == errCompactionsDisabled || err == errCompactionInProgress {
+							e.logger.Printf("aborted %s compaction group (%d). %v",
+								logDesc, groupNum, err)
+
+							if err == errCompactionInProgress {
+								time.Sleep(time.Second)
+							}
+							return
+						} else if err != nil {
 							e.logger.Printf("error compacting TSM files: %v", err)
 							atomic.AddInt64(&e.stats.TSMOptimizeCompactionErrors, 1)
 
@@ -950,7 +974,15 @@ func (e *Engine) compactTSMFull() {
 						}
 					} else {
 						files, err = e.Compactor.CompactFull(group)
-						if err != nil && err != errCompactionsDisabled {
+						if err == errCompactionsDisabled || err == errCompactionInProgress {
+							e.logger.Printf("aborted %s compaction group (%d). %v",
+								logDesc, groupNum, err)
+
+							if err == errCompactionInProgress {
+								time.Sleep(time.Second)
+							}
+							return
+						} else if err != nil {
 							e.logger.Printf("error compacting TSM files: %v", err)
 							atomic.AddInt64(&e.stats.TSMFullCompactionErrors, 1)
 

